### PR TITLE
[ENTESB-18654] wrong XSD url generated with XML initialiser

### PR DESCRIPTION
**Pull request by rnetuka**
Issue: https://issues.redhat.com/browse/ENTESB-18654

Upstream PRs:
https://github.com/apache/camel-k/pull/3101 (release-1.8.x)
https://github.com/apache/camel-k/pull/3102 (main)

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Comment by oscerd** [comment]
Please open a PR upstream too.

File: deploy/templates/xml.tmpl [hunk 1]
```diff
@@ -4,8 +4,8 @@
 <routes xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xmlns="http://camel.apache.org/schema/spring"
         xsi:schemaLocation="
-            http://camel.apache.org/schema/spring
-            http://camel.apache.org/schema/spring/camel-spring.xsd">
+            https://camel.apache.org/schema/spring
```
Comment (apupier) [inline deploy/templates/xml.tmpl]: this need to be kept without the https, it is the namespace. This cannto be changed withotu breaking backward compatibility.
it is only the xsd location which must be updated

**Review by apupier — DISMISSED** [review]

File: deploy/templates/xml.tmpl (hunk 1 above)
Comment (rnetuka) [reply deploy/templates/xml.tmpl]: Fixed. Please, check again.

**Review by apupier — COMMENTED** [review]
there are other places where the http is used instead of https. 

Do you prefer to modify in the same PR or create a different issue to handle it?
https://github.com/jboss-fuse/camel-k/blob/6b0174da9247ec609b4d70d712f485b8fbbcecc7/e2e/common/languages/files/xml.xml#L20
https://github.com/jboss-fuse/camel-k/blob/6b0174da9247ec609b4d70d712f485b8fbbcecc7/examples/polyglot/routes.xml#L20
https://github.com/jboss-fuse/camel-k/blob/6b0174da9247ec609b4d70d712f485b8fbbcecc7/pkg/util/modeline/parser_test.go#L147

**Comment by rnetuka** [comment]
@apupier I'll modify this PR. Thanks.

**Pull request merged**